hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/downloader/event/EventFileDownloader.java

```diff
@@ -59,34 +61,33 @@ public void download() {
     }
 
     /**
-     * Checks that hash of data file matches the verified hash and that data file is next in line based on previous file
-     * hash. Returns false if any condition is false.
+     * Reads the event file and checks that the file hash matches the verified hash and that data file is next in
+     * line based on previous file hash.
+     * @param file event file object
+     * @param verifiedHash the verified hash in hex
+     * @return StreamFile object
      */
     @Override
-    protected boolean verifyDataFile(File file, byte[] verifiedHash) {
+    protected StreamFile readAndVerifyDataFile(File file, String verifiedHash) {
         String expectedPrevFileHash = applicationStatusRepository.findByStatusCode(lastValidDownloadedFileHashKey);
         String fileName = file.getName();
+        Instant verifyHashAfter = downloaderProperties.getMirrorProperties().getVerifyHashAfter();
 
-        try {
-            EventFile eventFile = eventFileReader.read(file);
+        EventFile eventFile = eventFileReader.read(file);
 
-            if (!verifyHashChain(eventFile.getPreviousHash(), expectedPrevFileHash,
-                    downloaderProperties.getMirrorProperties().getVerifyHashAfter(), fileName)) {
-                log.error("PreviousHash mismatch for file {}. Expected = {}, Actual = {}", fileName,
-                        expectedPrevFileHash, eventFile.getPreviousHash());
-                return false;
-            }
+        if (!verifyHashChain(eventFile.getPreviousHash(), expectedPrevFileHash, verifyHashAfter, fileName)) {
+            throw new HashMismatchException(fileName, expectedPrevFileHash, eventFile.getPreviousHash());
+        }
 
-            String expectedFileHash = Hex.encodeHexString(verifiedHash);
-            if (!eventFile.getFileHash().contentEquals(expectedFileHash)) {
-                log.error("File {}'s hash mismatch. Expected = {}, Actual = {}", fileName, expectedFileHash,
-                        eventFile.getFileHash());
-                return false;
-            }
-        } catch (ImporterException e) {
-            log.error(e);
-            return false;
+        if (!eventFile.getFileHash().contentEquals(verifiedHash)) {
+            throw new HashMismatchException(fileName, expectedPrevFileHash, eventFile.getPreviousHash());
```

Comment:
Should be current hashes not previous
```suggestion
            throw new HashMismatchException(fileName, verifiedHash, eventFile.getFileHash());
```

Reply:
good catch. will correct it.